src/preprocessing/point_in_poly/winding_number_jacobson.jl

```diff
@@ -0,0 +1,136 @@
+struct NaiveWinding end
+
+@inline function (winding::NaiveWinding)(mesh, query_point)
+    (; face_vertices) = mesh
+
+    return naive_winding(mesh, face_vertices, query_point)
+end
+
+@inline function naive_winding(mesh, faces, query_point)
+    winding_number = sum(faces, init=0.0) do face
+
+        # A. Van Oosterom 1983,
+        # The Solid Angle of a Plane Triangle (doi: 10.1109/TBME.1983.325207)
+        a = face_vertex(mesh, face, 1) - query_point
+        b = face_vertex(mesh, face, 2) - query_point
+        c = face_vertex(mesh, face, 3) - query_point
+        a_ = norm(a)
+        b_ = norm(b)
+        c_ = norm(c)
+
+        divisor = a_ * b_ * c_ + dot(a, b) * c_ + dot(b, c) * a_ + dot(c, a) * b_
+
+        return 2atan(det([a b c]), divisor)
+    end
+
+    return winding_number
+end
+
+"""
+    WindingNumberJacobson(; shape=nothing, winding_number_factor=sqrt(eps()),
+                          hierarchical_winding=false)
+Algorithm for inside-outside segmentation of a complex shape proposed by Jacobson et al. (2013).
+
+# Keywords
+- `shape`: Complex shape returned by [`load_shape`](@ref) and is only required when using
+           `hierarchical_winding=true`.
+- `hierarchical_winding`: If set to `true`, an optimised hierarchical approach will be used
+                          which gives a significant speedup.
+                          It is only supported for 3D shapes yet.
+- `winding_number_factor`: For leaky shapes, a factor of `0.4` will give a better inside-outside segmentation.
+"""
+struct WindingNumberJacobson{ELTYPE}
+    winding_number_factor :: ELTYPE
+    winding               :: Union{NaiveWinding, HierarchicalWinding}
+
+    function WindingNumberJacobson(; shape=nothing, winding_number_factor=sqrt(eps()),
```

Comment:
```suggestion
                          It is only supported for 3D geometries yet.
- `winding_number_factor`: For leaky geometries, a factor of `0.4` will give a better inside-outside segmentation.
"""
struct WindingNumberJacobson{ELTYPE}
    winding_number_factor :: ELTYPE
    winding               :: Union{NaiveWinding, HierarchicalWinding}

    function WindingNumberJacobson(; geometry=nothing, winding_number_factor=sqrt(eps()),
```
As discussed earlier, please rename shape to geometry everywhere, except for the `ComplexShape`.